src/wasm/wasm-emscripten.cpp

```diff
@@ -739,6 +739,43 @@ void printSignatures(std::ostream& o, const std::set<Signature>& c) {
   o << "]";
 }
 
+static bool mainReadsParams(Module& module) {
+  // In the new llvm name mangling scheme for `main` we get the
+  // 2 argument main mangled as `__main_argc_argv` and the no
+  // argument form generates and alias called `__main_void`
+  if (module.getExportOrNull("__main_void")) {
+    return false;
```

Comment:
why return false here? Comment might help understand why we don't need to do anything.

Reply:
The comment above is supposed to describe that but I can add more.   Presence of `__main_void` as an export means main does not take args.